hr_holidays_credit/tests/test_hr_holidays_credit.py

```diff
@@ -0,0 +1,274 @@
+            }
+        )
+        allocation.action_confirm()
+        allocation.action_validate()
+
+        with self.assertRaises(ValidationError):
+            self.SudoLeave.create(
+                {
+                    "holiday_status_id": leave_type.id,
+                    "holiday_type": "employee",
+                    "employee_id": employee.id,
+                    "date_from": "2020-01-01",
+                    "date_to": "2020-01-10",
+                }
+            )
+
+        leave_type.write({"allow_credit": True})
+        self.SudoLeave.create(
+            {
+                "holiday_status_id": leave_type.id,
+                "holiday_type": "employee",
+                "employee_id": employee.id,
+                "date_from": "2020-01-01",
+                "date_to": "2020-01-10",
+            }
+        )
+
+    def test_2(self):
+        department = self.SudoDepartment.create({"name": "Department #2"})
+        employee_1 = self.SudoEmployee.create(
+            {"name": "Employee #2-1", "department_id": department.id}
+        )
+        employee_2 = self.SudoEmployee.create(
+            {"name": "Employee #2-2", "department_id": False}
+        )
+
+        leave_type = self.SudoLeaveType.create(
+            {
+                "name": "Leave Type #2",
+                "requires_allocation": "yes",
+                "allocation_validation_type": "officer",
+                "allow_credit": True,
+                "creditable_department_ids": [(6, False, [department.id])],
+            }
+        )
+
+        allocation_1 = self.SudoAllocation.create(
+            {
+                "holiday_status_id": leave_type.id,
+                "number_of_days": 5,
+                "employee_id": employee_1.id,
+                "date_from": "2020-01-01",
+                "date_to": "2020-12-31",
+            }
+        )
+        allocation_2 = self.SudoAllocation.create(
+            {
+                "holiday_status_id": leave_type.id,
+                "number_of_days": 5,
+                "employee_id": employee_2.id,
+                "date_from": "2020-01-01",
+                "date_to": "2020-12-31",
+            }
+        )
+
+        allocation_1.action_confirm()
+        allocation_1.action_validate()
+        allocation_2.action_confirm()
+        allocation_2.action_validate()
+
+        self.SudoLeave.create(
+            {
+                "holiday_status_id": leave_type.id,
+                "holiday_type": "employee",
+                "employee_id": employee_1.id,
+                "number_of_days": 1,
+            }
+        )
+        with self.assertRaises(ValidationError):
+            self.SudoLeave.create(
+                {
+                    "holiday_status_id": leave_type.id,
+                    "holiday_type": "employee",
+                    "employee_id": employee_2.id,
+                    "number_of_days": 1,
+                }
+            )
+
+    def test_3(self):
```

Comment:
"""
        Test that leave allocation validation respects individual employee credit restrictions.
        Leaves for employees with credit allowed succeed, while others raise ValidationError.
        """